src/lxml/serializer.pxi

```diff
@@ -1035,3 +1044,22 @@ cdef class _FileWriterElement:
     def __exit__(self, exc_type, exc_val, exc_tb):
         self._writer._write_end_element(self._element)
         self._writer._method = self._old_method
+
+@cython.final
+@cython.internal
+@cython.freelist(8)
+cdef class _MethodChanger:
+    cdef int _new_method
+    cdef int _old_method
+    cdef _IncrementalFileWriter _writer
+
+    def __cinit__(self, _IncrementalFileWriter writer not None, int method):
+        self._writer = writer
+        self._new_method = method
+        self._old_method = writer._method
```

Comment:
I guess I'm among the guiltiest when it comes to using context managers outside of with statement. Look for ```__enter__``` and ```__exit__``` cals in: https://github.com/arskom/spyne/blob/f92823a512eb8be71adb24fa151b397f27df5b9f/spyne/protocol/cloth/to_cloth.py#L278

To me the contract with using context managers' private apis is it's user's responsibility to enter/exit in correct order. I could totally take the precautions you mention but aren't they needless overhead for people who use the apis the correct way (within the with statement) or incorrect way (like me) but who are supposed to be people who know what they are doing?

Maybe in the spirit of protecting the expert, we could set a flag on the context manager to prevent it from being __exit__'ed twice.

However if you insist, I will implement the stack.